### PR TITLE
Modify rule S6317: Update to LayC format (APPSEC-968)

### PR DESCRIPTION
**Pull request by egon-okerman-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S6317/common/fix/least-privilege.adoc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+==== Principle of least privilege
+When creating IAM policies, it is important to adhere to the principle of least privilege. This means that any user or role should only be granted enough permissions to perform the tasks that they are supposed to, and __nothing else__.
```
Comment (sebastien-andrivet-sonarsource) [inline rules/S6317/common/fix/least-privilege.adoc:2]: It is a detail, but I am not sure when we put an empty line after the subsection, and when we do not. This does not look consistent but maybe it has no impact on what the users see.

File: rules/S6317/terraform/how-to-fix-it/aws-iam.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
+== How to fix it in AWS Identity and Access Management
+
+=== Code examples
+
+In this example, the IAM policy allows an attacker to update the code of any Lambda function. An attacker can achieve privilege escalation by altering the code of a Lambda that executes with high privileges.
+
+==== Noncompliant code example
+
+[source,terraform]
```
Comment (sebastien-andrivet-sonarsource) [inline rules/S6317/terraform/how-to-fix-it/aws-iam.adoc:9]: The diff attributes are missing

**Review by sebastien-andrivet-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S6317/metadata.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "title": "AWS IAM policies should not allow privilege escalation",
+  "title": "AWS IAM policies should limit the scope of permissions given",
   "type": "VULNERABILITY",
   "code": {
     "impacts": {
```
[the remark this answers is not in the document]
Comment (egon-okerman-sonarsource) [reply rules/S6317/metadata.json:5]: I don't think Conventional is the best attribute here, but then again I do not know which one would fit better.

@hendrik-buchwald-sonarsource do you have any thoughts on this?

[the remark this answers is not in the document]
Comment (hendrik-buchwald-sonarsource) [reply rules/S6317/metadata.json:5]: Based on the [decision tree](https://xtranet-sonarsource.atlassian.net/wiki/spaces/PM/pages/2869067851/LaYC+Classifying+Rules), I think it is conventional. It is not a perfect fit but I don't think we should overthink it for now.

**Review by sebastien-andrivet-sonarsource — APPROVED** [review]
LGTM, but should probably be rebased on master because of a dead link.

**Pull request merged**